### PR TITLE
Remove unnecessary input for different restore mode

### PR DESCRIPTION
**Pull request by pixia**
## Description

We expect different parameters in different restore mode. We have been passing unnecessary parameters in restore request. This changes removes those unnecessary parameters.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Sql/Sql/ChangeLog.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 -->
 ## Upcoming Release
 * Add support for SQL DB Hyperscale tier
+* Removed unnecessary parameters passed in different restore mode
```
Comment (jaredmoo) [inline src/Sql/Sql/ChangeLog.md]: This makes it sound like the powershell command line parameters were changed, which sounds a bit scary. Can you rephrase to state that it was a bug fix? e.g. "Fixed bug where restore could fail due to setting unnecessary properties in restore request"

Comment (pixia) [reply src/Sql/Sql/ChangeLog.md]: updated.

**Comment by maddieclayton** [comment]
@azuresdkci Add to whitelist
